webapp/src/components/sidebar_right/index.tsx

```diff
@@ -51,31 +51,51 @@ export const renderThumbVertical = (props: Props) => (
     />
 );
 
+// React hook to store the last state of "yourPrs" and "reviews"
+function usePrevious(value: Item[]) {
+    const ref = useRef<Item[]>();
+
+    useEffect(() => {
+      ref.current = value;
+    }, [value]);
+
+    return ref.current;
+}
+
+function shouldUpdateDetails(prs: Item[], prevPrs: Item[], targetState: string, currentState: string) {
+    if (currentState !== targetState) {
+        return false;
+    }
+
+    if (prs.length !== prevPrs.length) {
+        return true;
+    }
+
+    for (let i = 0; i < prs.length; i++) {
+        if (prs[i].id !== prevPrs[i].id) {
+            return true;
+        }
+    }
+    return false;
+}
+
 function SidebarRight({theme}: {theme: Theme}) {
     const sidebarData = useSelector(getSidebarData);
     const {username, yourAssignments, org, unreads, gitlabURL, rhsState, reviews, yourPrs} = sidebarData;
 
     const dispatch = useDispatch();
 
-    // Get the details for PRs and reviews on the first render
-    useEffect(() => {
-        if (yourPrs && rhsState === RHSStates.PRS) {
-            dispatch(getYourPrDetails(yourPrs));
-        }
-
-        if (reviews && rhsState === RHSStates.REVIEWS) {
-            dispatch(getReviewDetails(reviews));
-        }
-    }, []);
+    const prevPrs = usePrevious(yourPrs)
+    const prevReviews = usePrevious(reviews)
 
     useEffect(() => {
-        if (RHSStates.PRS === rhsState) {
+        if (yourPrs && (!prevPrs || shouldUpdateDetails(yourPrs, prevPrs, RHSStates.PRS, rhsState))) {
             dispatch(getYourPrDetails(yourPrs));
         }
     }, [yourPrs, rhsState]);
 
     useEffect(() => {
-        if (RHSStates.REVIEWS === rhsState) {
+        if (reviews && (!prevReviews || shouldUpdateDetails(reviews, prevReviews, RHSStates.REVIEWS, rhsState))) {
             dispatch(getReviewDetails(reviews));
         }
     }, [reviews, rhsState]);
```

Comment:
Do we also need to provide `prevReviews` as a hook dependency here?

Reply:
@mickmister Same in this case, `prevReviews` will only change when `reviews` is changed and that is already present in the dependency array.